modules/platforms/cpp/thin-client/include/ignite/thin/cache/event/java_cache_entry_event_filter.h

```diff
@@ -0,0 +1,129 @@
+ */
+
+/**
+ * @file
+ * Declares ignite::thin::cache::event::JavaCacheEntryEventFilter class.
+ */
+
+#ifndef _IGNITE_THIN_CACHE_EVENT_JAVA_CACHE_ENTRY_EVENT_FILTER
+#define _IGNITE_THIN_CACHE_EVENT_JAVA_CACHE_ENTRY_EVENT_FILTER
+
+#include <ignite/thin/cache/event/cache_entry_event.h>
+
+#include <ignite/impl/thin/platform_java_object_factory_proxy.h>
+
+namespace ignite
+{
+    namespace impl
+    {
+        namespace thin
+        {
+            namespace cache
+            {
+                class CacheClientImpl;
+            }
+        }
+    }
+
+    namespace thin
+    {
+        namespace cache
+        {
+            namespace event
+            {
+                /**
+                 * Java cache entry event filter.
+                 *
+                 * All templated types should be default-constructable,
+                 * copy-constructable and assignable.
+                 */
+                class JavaCacheEntryEventFilter
+                {
+                    friend ignite::impl::thin::cache::CacheClientImpl;
+                public:
+                    /**
+                     * Default constructor.
+                     */
+                    JavaCacheEntryEventFilter()
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Constructor.
+                     *
+                     * @param factoryClassName Name of the Java factory class.
+                     */
+                    JavaCacheEntryEventFilter(const std::string& factoryClassName) :
+                        factory(new impl::thin::PlatformJavaObjectFactoryProxy(
+                            impl::thin::FactoryType::USER, factoryClassName))
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Destructor.
+                     */
+                    virtual ~JavaCacheEntryEventFilter()
+                    {
+                        // No-op.
+                    }
+
+                    /**
+                     * Get Java remote filter factory class name.
+                     *
+                     * @return Java factory class name.
+                     */
+                    const std::string& GetFactoryClassName() const
+                    {
+                        return factory.Get()->GetFactoryClassName();
+                    }
+
+                    /**
+                     * Add property.
+                     *
+                     * Template argument type should be copy-constructable and assignable. Also BinaryType class template
+                     * should be specialized for this type.
+                     *
+                     * @param name Property name.
+                     * @param value Property value.
+                     */
+                    template<typename T>
+                    void SetProperty(const std::string& name, const T& value)
```

Comment:
We can set properties, but not get them - is this intentional?

Reply:
Well, yeah. I can add a method for getting property, but it will be hard to implement right and I don't think there are a lot of use cases where we'd like to get properties from such an object anyway.